v2/internal/util/cel/cel_test.go

```diff
@@ -0,0 +1,553 @@
+			secret: map[string]string{
+				"pizza": "pepperoni",
+			},
+			expression:  `self.spec.location + " " + secret.pizza`,
+			expectedStr: "eastus pepperoni",
+		},
+		{
+			name: "missing secret value",
+			self: newSimpleResource(),
+			secret: map[string]string{
+				"pizza": "pepperoni",
+			},
+			expression:  `self.spec.location + " " + secret.sauce`,
+			expectedErr: `failed to eval CEL expression: "self.spec.location + \" \" + secret.sauce": no such key: sauce`,
+		},
+		{
+			name: "underscore field works as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.UnderscoreField = "hello"
+				}),
+			expression:  `self.spec.underscore_field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "double underscores field escaped as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.DoubleUnderscoreField = "hello"
+				}),
+			expression:  `self.spec.underscore__underscores__field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "double double underscores field escaped as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.DoubleDoubleUnderscoreField = "hello"
+				}),
+			expression:  `self.spec.underscore__underscores__field__underscores__`,
+			expectedStr: "hello",
+		},
+		{
+			name: "underscore start field works as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.UnderscoreStartField = "hello"
+				}),
+			expression:  `self.spec._underscoreStartField`,
+			expectedStr: "hello",
+		},
+		{
+			name: "dash field escaped as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.DashField = "hello"
+				}),
+			expression:  `self.spec.dash__dash__field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "dash + underscore field escaped as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.DashUnderscoreField = "hello"
+				}),
+			expression:  `self.spec.dash__dash____underscores__field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "dot field escaped as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.DotField = "hello"
+				}),
+			expression:  `self.spec.dot__dot__field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "slash field works as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.SlashField = "hello"
+				}),
+			expression:  `self.spec.slash__slash__field`,
+			expectedStr: "hello",
+		},
+		{
+			name: "reserved fields works as expected",
+			self: newSimpleResource2Customized(
+				func(r *SimpleResource2) {
+					r.Spec.Const = "hello"
+					r.Spec.Break = " there"
+					r.Spec.Loop = " friend"
+				}),
+			expression:  `self.spec.const + self.spec.break + self.spec.loop`,
+			expectedStr: "hello there friend",
+		},
+	}
+
+	for _, c := range cases {
+		c := c
```

Comment:
YIL (yesterday-I-learned) that this is no longer needed; as of [Go 1.22](https://tip.golang.org/doc/go1.22#language), loop variables are created new for each iteration.

Reply:
I actually "knew" this but hadn't really registered that it meant I could avoid it in ASO.

Removed!